### PR TITLE
Catch various 404s from message operations

### PR DESCRIPTION
**Pull request by MarkKoz**
Fixes Sentry issues [BOT-ZN](https://sentry.io/organizations/python-discord/issues/2385622541), [BOT-J2](https://sentry.io/organizations/python-discord/issues/2091776305/?referrer=github_integration), and [BOT-1J7](https://sentry.io/organizations/python-discord/issues/2529286507/?referrer=github_integration)

### CODE REVIEW

**Review by Xithrius — APPROVED** [review]

File: bot/exts/moderation/incidents.py [hunk 1]
```diff
@@ -143,7 +143,14 @@ async def add_signals(incident: discord.Message) -> None:
             log.trace(f"Skipping emoji as it's already been placed: {signal_emoji}")
         else:
             log.trace(f"Adding reaction: {signal_emoji}")
-            await incident.add_reaction(signal_emoji.value)
+            try:
+                await incident.add_reaction(signal_emoji.value)
+            except discord.NotFound as e:
+                if e.code != 10008:
+                    raise
```
[the remark this answers is not in the document]
Comment (MarkKoz) [reply bot/exts/moderation/incidents.py:150]: It's not required, and doing so actually causes the `raise e` line to be included in the traceback.

File: bot/exts/fun/duck_pond.py [hunk 2]
```diff
@@ -171,8 +171,14 @@ async def on_raw_reaction_add(self, payload: RawReactionActionEvent) -> None:
         if not self.is_helper_viewable(channel):
             return
 
-        message = await channel.fetch_message(payload.message_id)
+        try:
+            message = await channel.fetch_message(payload.message_id)
+        except discord.NotFound:
+            return  # Message was deleted.
+
```
[the remark this answers is not in the document]
Comment (MarkKoz) [reply bot/exts/fun/duck_pond.py:178]: Sometimes. It's not consistent nor something that's being enforced.

[the remark this answers is not in the document]
Comment (Bluenix2) [reply bot/exts/fun/duck_pond.py:178]: Well here I would say it makes sense?

[the remark this answers is not in the document]
Comment (MarkKoz) [reply bot/exts/fun/duck_pond.py:178]: I don't really think this is much worse; it's not worth it for me to make a new commit to change this at this point. I have mixed feelings on it anyway, but I'm not gonna get into that if I don't have to.

**Pull request merged**